### ISSUE TITLE
WRONG_INDENTATION: aligned chained method calls reported as false positives by the linter

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [saveourtool/diktat#1336]
Consider this code example:

```kotlin
fun LocalDateTime.updateTime(
    hour: Int? = null,
    minute: Int? = null,
    second: Int? = null,
): LocalDateTime = withHour(hour ?: getHour())
    .withMinute(minute ?: getMinute())
    .withSecond(second ?: getSecond())
```

Chained method calls in lines 6-7 are normally aligned. Yet, DiKTat reports this as an error, suggesting the following formatting. `extendedIndentBeforeDot` **on** (the default):

```kotlin
fun LocalDateTime.updateTime(
    hour: Int? = null,
    minute: Int? = null,
    second: Int? = null,
): LocalDateTime = withHour(hour ?: getHour())
        .withMinute(minute ?: getMinute())
                .withSecond(second ?: getSecond())
```

&mdash; and `extendedIndentBeforeDot` **off**:

```kotlin
fun LocalDateTime.updateTime(
    hour: Int? = null,
    minute: Int? = null,
    second: Int? = null,
): LocalDateTime = withHour(hour ?: getHour())
    .withMinute(minute ?: getMinute())
        .withSecond(second ?: getSecond())
```

**Comment by 0x6675636b796f75676974687562** [comment]
A prerequisite for this is an expression-body function, i. e. none of the below will trigger the issue:

```kotlin
val a = expression

val b =
        expression

val c
    get() =
            expression

fun d() {
    expression
}
```

while this _may_:

```kotlin
fun e() =
        expression
```

Minimal repro (assuming `extendedIndentBeforeDot` is **on**):

```kotlin
fun f() = first()
        .second()
        .third() // <- here, DiKTat suggest a double continuation indent (16) instead of a single continuation indent.
```

At the same time, this fragment produces no warnings (which is correct):

```kotlin
fun f() =
        first() // Continuation indent
                .second() // Double continuation indent
                .third() // Double continuation indent aligned
```

Another minimal repro:

```kotlin
fun f() = g(first()
        .second() // DiKTat suggest an indentation of 12
        .third() // DiKTat suggest an indentation of 20
        .fourth()) // DiKTat suggest an indentation of 28
```

Similarly, this fragment produces no warnings (which is correct, too):

```kotlin
fun f() = g(
        first()
                .second()
                .third()
                .fourth())
```
